Unexpected Missing-Docs on extension declaration

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When defining an extension of a Class the `missing_docs` triggers a warning even if the Class definition is documented.

I think we could check and avoid triggering this error on extension declarations, since the class itself will trigger the warning if there is no doc.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'main.swift' (1/1)
/Users/carlos/Projects/SwiftLinter/TestProject/main.swift:5:8: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.34.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? HomeBrew
* Paste your configuration file:

```yml
opt_in_rules: # some rules are only opt-in
   - missing_docs
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? no
* Which Xcode version are you using? Version 10.2.1 (10E1001)
* Do you have a sample that shows the issue? 

echo -e "///Docs\npublic class Foo {\n}\n public extension Foo {\n} " | swiftlint lint --no-cache --use-stdin --enable-all-rules

```swift
/// Foo Docs
public class Foo {
}

public extension Foo {
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -56,6 +56,10 @@
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#2842](https://github.com/realm/SwiftLint/issues/2842)
 
+* Don't trigger `missing_docs` violations on extensions.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2851](https://github.com/realm/SwiftLint/issues/2851)
+
 ## 0.34.0: Anti-Static Wool Dryer Balls
 
 #### Breaking

--- a/Rules.md
+++ b/Rules.md
@@ -12008,6 +12008,10 @@ public class A {
 }
 ```
 
+```swift
+public extension A {}
+```
+
 </details>
 <details>
 <summary>Triggering Examples</summary>

--- a/Source/SwiftLintFramework/Rules/Lint/MissingDocsRule.swift
+++ b/Source/SwiftLintFramework/Rules/Lint/MissingDocsRule.swift
@@ -10,7 +10,10 @@ private extension File {
         let substructureOffsets = dictionary.substructure.flatMap {
             missingDocOffsets(in: $0, acls: acls)
         }
+        let extensionKinds: Set<SwiftDeclarationKind> = [.extension, .extensionEnum, .extensionClass,
+                                                         .extensionStruct, .extensionProtocol]
         guard let kind = dictionary.kind.flatMap(SwiftDeclarationKind.init),
+            !extensionKinds.contains(kind),
             case let isDeinit = kind == .functionMethodInstance && dictionary.name == "deinit",
             !isDeinit,
             let offset = dictionary.offset,
@@ -54,6 +57,9 @@ public struct MissingDocsRule: OptInRule, ConfigurationProviderRule, AutomaticTe
             public class A {
                 deinit {}
             }
+            """,
+            """
+            public extension A {}
             """
         ],
         triggeringExamples: [

